services/venice-server/src/main/java/com/linkedin/venice/listener/HttpChannelInitializer.java

```diff
@@ -68,20 +71,35 @@ public class HttpChannelInitializer extends ChannelInitializer<SocketChannel> {
   private final ServerLoadControllerHandler loadControllerHandler;
   private boolean isDaVinciClient;
 
+  // Dependencies needed for gRPC request processor in test mode
+  private final StorageEngineRepository storageEngineRepository;
+  private final ReadOnlyStoreRepository storeMetadataRepository;
+  private final ReadOnlySchemaRepository schemaRepository;
+  private final StorageEngineBackedCompressorFactory compressorFactory;
+
   public HttpChannelInitializer(
+      StorageEngineRepository storageEngineRepository,
```

Comment:
Instead of changing constructor of `HttpChannelInitializer`, we could explore following the previous implementation of GrpcStorageReadRequestHandler by adding a new request type of CountByType and its handler function inside `StorageReadRequestHandler`. Inside `StorageReadRequestHandler`, there are exiting processCompute, handleComputeRequest function you could leverage.